### PR TITLE
Add Date widget

### LINKED ISSUE #58: Implement Date widget

### PR DESCRIPTION
**Pull request by anserwaseem**
## Issue ticket number and link

Closes #58

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests

### CODE REVIEW

File: packages/runtime/src/widgets/Date.tsx [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+
+  const { values } = useRegisterBindings({ ...props, value }, props.id, {
+    setValue,
+  });
+
+  const action = useEnsembleAction(props.onChange);
+  const onChangeCallback = useCallback(
+    (date: any) => {
+      const formattedDate = dayjs(toString(date))?.format(DateDisplayFormat);
+      formattedDate !== "Invalid Date" && setEnteredDate(formattedDate);
```
Comment (evshi) [inline packages/runtime/src/widgets/Date.tsx:74]: This should be a separate hook to sync the date with state. Hooks should be responsible for one thing

File: packages/runtime/src/widgets/Date.tsx [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+          />
+        </LocalizationProvider>
+      </AntForm.Item>
+    </DatePickerContext.Provider>
+  );
+};
+
+WidgetRegistry.register("Date", Date);
+
+const CustomCalendarHeader: React.FC<PickersCalendarHeaderProps<any>> = (
```
Comment (evshi) [inline packages/runtime/src/widgets/Date.tsx:207]: Why is this extra component needed?

File: packages/runtime/src/widgets/Date.tsx [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+              margin: "24px",
+            }}
+          />
+        )}
+      </div>
+    </>
+  );
+};
+
+const CustomActionBar: React.FC<PickersActionBarProps> = (props) => {
```
Comment (evshi) [inline packages/runtime/src/widgets/Date.tsx:314]: Why is this extra component needed?

File: packages/runtime/src/widgets/Date.tsx (hunk 2 above)
Comment (anserwaseem) [reply packages/runtime/src/widgets/Date.tsx:207]: these extra components are being used here https://github.com/EnsembleUI/ensemble-react/blob/d6c200c43766f6c2db8c3064d8863df8e0fbd3f4/packages/runtime/src/widgets/Date.tsx#L130-L133

**Review by evshi — APPROVED** [review]

**Pull request merged**